tests/src/test/java/alluxio/client/FileInStreamIntegrationTest.java

```diff
@@ -425,48 +424,77 @@ public void positionedReadWithoutCaching() throws Exception {
   }
 
   @Test(timeout = 10000)
-  @Ignore
   public void asyncCacheFirstBlock() throws Exception {
     String filename = mTestPath + "/file_" + MAX_LEN + "_" + mWriteUnderStore.hashCode();
     AlluxioURI uri = new AlluxioURI(filename);
 
-    FileInStream is =
-        mFileSystem.openFile(uri, OpenFileOptions.defaults().setReadType(ReadType.CACHE));
-    is.read();
-    URIStatus status = mFileSystem.getStatus(uri);
-    Assert.assertEquals(0, status.getInAlluxioPercentage());
-    is.close();
-    CommonUtils.waitFor("First block to be cached.", (input) -> {
-      try {
-        URIStatus st = mFileSystem.getStatus(uri);
-        return !st.getFileBlockInfos().get(0).getBlockInfo().getLocations().isEmpty();
-      } catch (Exception e) {
-        return false;
+    for (ReadType readType : ReadType.values()) {
+      // Prepare: ensure no in-Alluxio data from previous iteration
+      mFileSystem.free(uri);
+      CommonUtils.waitFor("No in-Alluxio data left from previous iteration.", (input) -> {
+        try {
+          URIStatus st = mFileSystem.getStatus(uri);
+          return st.getInAlluxioPercentage() == 0;
+        } catch (Exception e) {
+          return false;
+        }
+      });
+      FileInStream is = mFileSystem.openFile(uri, OpenFileOptions.defaults().setReadType(readType));
+      is.read();
+      URIStatus status = mFileSystem.getStatus(uri);
+      Assert.assertEquals(0, status.getInAlluxioPercentage());
+      is.close();
+      if (readType.isCache()) {
+        CommonUtils.waitFor("First block to be cached.", (input) -> {
+          try {
+            URIStatus st = mFileSystem.getStatus(uri);
+            return !st.getFileBlockInfos().get(0).getBlockInfo().getLocations().isEmpty();
```

Comment:
Thinking about this test again, we should also test the negative cases, ie. the other block infos have empty locations.

Reply:
fixed